daffodil-core/src/test/scala/org/apache/daffodil/core/grammar/primitives/TestPrimitives.scala

```diff
@@ -62,31 +64,36 @@ class TestPrimitives {
     val optMatch = re.findFirstMatchIn("'P'###012V34'P';N#N")
     println(optMatch)
     optMatch match {
-      case Some(re("'P'", "###012", "34", "'P'", "N", "#", "N")) => // ok
+      case Some(m) if m.subgroups == List("'P'", "###012", "34", "'P'", "N", "#", "N") => // ok
+      case _ => fail()
     }
   }
 
   @Test def testVRegexOnlyPositivePattern(): Unit = {
     val re = TextNumberPatternUtils.vRegexStandard
     val Some(myMatch) = re.findFirstMatchIn("A###012V34B")
-    myMatch match {
-      case re("A", "###012", "34", "B", null, null, null) =>
+    myMatch.subgroups match {
+      case List("A", "###012", "34", "B", null, null, null) => // ok
+      case _ => fail()
     }
```

Comment:
Why did this test and other tests have to change when others at the top of this file look exactly the same but didn't need changes? Does it have to do with the null's or somthing?

I would also suggest making yoru changes consistent. Sometimes you do `Some(m) if m.subgroups`, sometimes you do `val Some(myMatch) = ...` with `case myMatch.subgroups`. If we have to change these tests we should try to do so in a consistent way.